File: content-solutions/hash/camping-gear.java
import java.util.*;

class Solution {
    public int solution(String[] kinds) {
        Map<String, Integer> count = new HashMap<>();
        for (String kind : kinds) count.put(kind, count.getOrDefault(kind, 0) + 1);
        int ways = 1;
        for (int c : count.values()) {
            ways *= c + 1;
        }
        return ways - 1;
    }
}

File: content-solutions/hash/camping-gear.js
function solution(kinds) {
  const count = new Map();
  for (const kind of kinds) count.set(kind, (count.get(kind) ?? 0) + 1);
  let ways = 1;
  for (const c of count.values()) {
    ways *= c + 1;
  }
  return ways - 1;
}
